gadget/gadget.go

```diff
@@ -42,7 +42,7 @@ const validSnapIDLength = 32
 var (
 	validVolumeName = regexp.MustCompile("^[a-zA-Z0-9][a-zA-Z0-9-]+$")
 	validTypeID     = regexp.MustCompile("^[0-9A-F]{2}$")
-	validGUUID      = regexp.MustCompile("^[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}$")
+	validGUUID      = regexp.MustCompile("^([0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}|[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})$")
```

Comment:
silly question: why not simply `(?i)` it? Do we really care that it be all lowercase or all uppercase?

Reply:
We could, `ubuntu-image` kind of does that, but only because Python's UUID library does not care and converts the thing to a binary representation anyway. On the other hand, `00000000-0000-0000-abcd-dd00DeaDBeeF` looks a bit awkward.

Reply:
Mixed case hex could be unusual/ugly/awkward, but would it be **wrong**?

Reply:
https://youtu.be/k6SSPUcCln8?t=15

Reply:
Haha :smile: I'm ok with letting mixed casing, we can pacify it it elsewhere.